scripts/gen_lsp.lua

```diff
@@ -35,7 +35,10 @@ end
 
 -- Gets the Lua symbol for a given fully-qualified LSP method name.
 local function name(s)
-  return s:gsub('/', '_', 3)
+  if vim.startswith(s, '$') then
+    s = s:gsub('^%$', 'notification')
```

Comment:
Based on https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#dollarRequests it seems misleading to use `notification_` prefix for `$/` notifications. Because all other notifications _aren't_ prefixed with `$/`.

Instead maybe the prefix should be `optional_`. Or just `dollar_`.

Reply:
As a piece of data to help make your decision, in my LSP framework for Elixir I went with "dollar" for these

https://hexdocs.pm/gen_lsp/GenLSP.Notifications.DollarProgress.html

Reply:
`dollar` might be a good choice.

Reply:
Let's try that. @glepnir can you regen the file